citar.el

```diff
@@ -572,6 +615,35 @@ HISTORY is the 'completing-read' history argument."
          ((string-match "http" resource 0) "Links")
          (t "Library Files")))))
 
+(defun citar--ref-completion-table ()
+  "Return completion table for cite keys, as a hash table.
+In this hash table, keys are a strings with author, date, and
+title of the reference.  Values are the cite keys.
+Return nil if there are no bibliography files or no entries."
+  ;; Populate bibliography cache.
+  (when-let ((bibs (citar--bibliographies)))
+    (let* ((entries (citar-cache--entries bibs))
+           (preformatted (citar-cache--preformatted bibs))
+           (fmtstr (citar--get-template 'completion))
+           (hasnotep (citar-has-note))
+           (hasfilep (citar-has-file))
```

Comment:
I mentioned on #628 that I think we should rename this to `citar-has-library-file`; for clarity.

Reply:
Hmm, I agree with renaming it, but to me "library file" also seems very vague and generic. How about something like "attachment"? And would the "has:files" tag in the completion string also change? It is currently easy to type at least.

Reply:
> Hmm, I agree with renaming it, but to me "library file" also seems very vague and generic. How about something like "attachment"? 

"attachment" has too much association with emails, I think?

The "library" word would just be to be consistent with the defcustom etc.

But it's not a big deal, particularly since ...

> And would the "has:files" tag in the completion string also change? It is currently easy to type at least.

No; I would keep it as is, for the reason you note.